comp/core/secrets/secretsimpl/secrets.go

```diff
@@ -261,8 +365,12 @@ type secretInfo struct {
 }
 
 // GetDebugInfo exposes debug informations about secrets to be included in a flare
-func GetDebugInfo(w io.Writer) {
-	if secretBackendCommand == "" {
+func (r *secretResolver) GetDebugInfo(w io.Writer) {
+	if !r.enabled {
+		log.Errorf("Agent secrets is disabled by caller")
+		return
+	}
```

Comment:
We should not log an error but print to the `io.Writer` a message that the secrets feature is disabled.